pkg/auth/oidc.go

```diff
@@ -0,0 +1,110 @@
+package auth
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+var InvalidAuthError = status.Errorf(codes.Unauthenticated, "invalid authentication credentials")
+var OIDCHeaderKey = "X-OIDC-Email"
```

Comment:
let's make it configurable via `OIDCAuth.HeaderKey` instead of hardcoding it

Reply:
after giving it some thought, I only saw two usage of header key (containing email). First at `ctx_logrus.AddFields` [code](https://github.com/odpf/guardian/blob/main/internal/server/auth.go#L21). I think it's used for audit log. 

second, it is located at `getUser` [code](https://github.com/odpf/guardian/blob/main/api/handler/v1beta1/grpc.go#L154). I realized it because when rejecting an appeal, it still needs email on header request (if I choose OIDC auth)

as for first location, it is already solved by getting email from OIDC JWT payload, then set logrus fields key to `OIDCHeaderKey`. I think we can standardize this logrus field somehow to a default value (same with default auth logrus field)

as for second location, I propose that we set context value to `getUser` with two values (it will be set on interceptor)
- auth type
- context key
then we can get the user from context instead of getting it from metadata header request. How does that sound?

Reply:
interesting, for second location, we can also just use a single context value containing the authentication email (like the one in first location). And the interceptor will set the value according to the auth type accordingly. 

With this approach, we can also set the logrus field only once since the prior auth interceptor has set the authentication email context value like I mentioned above. What do you think?

Reply:
i think that's good. i will modify it shortly

Reply:
hi @rahmatrhd i have adjust the code, do you have any comments on that?